cirq-google/cirq_google/api/v2/program.proto

```diff
@@ -431,11 +431,31 @@ message ArgMapping {
   repeated ArgEntry entries = 1;
 }
 
+message FunctionInterpolationData {
+  // If the function encoded has dimension `d`, then |x| = d |y|. Where |x| the length of `x`.
+  repeated float x = 1 [packed = true];  // The independent variable.
+  repeated float y = 2 [packed = true];  // The dependent variable.
+
+  // The interpolation method.
+  // Currently only "interp" (i.e numpy.interp) is supported.
+  string method = 3;
+}
+
+message CustomArg {
+  oneof custom_arg {
+    FunctionInterpolationData function_interpolation_data = 1;
+  }
+}
+
 message InternalGate{
   string name = 1;  // Gate name.
   string module = 2;  // Gate module. 
   int32 num_qubits = 3;  // Number of qubits. Required during deserialization.
   map<string, Arg> gate_args = 4;  // Gate args.
+
+  // Custom args are arguments that need require special processing during
```

Comment:
How do `gate_args` differ from `custom_args`? i.e. could `FunctionInterpolationData` be added to the existing [`Arg`](https://github.com/quantumlib/Cirq/blob/599faf62c89d38068b4f80cdc5219162179fda76/cirq-google/cirq_google/api/v2/program.proto#L283) to support use of the existing `gate_args`? If not, is there a clear guidance (and documentation) for whether a new argument type should be an `Arg` vs a `CustomArg`?

Reply:
> How do gate_args differ from custom_args? 

`Arg` are values that we can use as is. That's after deserialization we can use that value internally. `CustomArgs` are args that we need to perform extra processing on before we can use. 

> could FunctionInterpolationData be added to the existing [Arg](https://github.com/quantumlib/Cirq/blob/599faf62c89d38068b4f80cdc5219162179fda76/cirq-google/cirq_google/api/v2/program.proto#L283) to support use of the existing gate_args?
no 

---

I think if we just add an extra possiblitiy to `Arg` we will just increase the complexity of (de)serialization with no obvious gain.